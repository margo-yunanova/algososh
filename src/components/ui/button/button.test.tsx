import renderer from "react-test-renderer";
import { render, screen, fireEvent } from "@testing-library/react";
import { Button } from "./button";

it("renders button with text", () => {
  const tree = renderer.create(<Button text="Развернуть" />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders button without text", () => {
  const tree = renderer.create(<Button text="" />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders disabled button", () => {
  const tree = renderer
    .create(<Button text="Развернуть" disabled={true} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders button with loader", () => {
  const tree = renderer
    .create(<Button text="Развернуть" isLoader={true} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("alert by push button", () => {
  window.alert = jest.fn();
  render(
    <Button text="Развернуть" onClick={() => alert("Button has pushed")} />
  );
  fireEvent.click(screen.getByText("Развернуть"));
  expect(window.alert).toHaveBeenCalledWith("Button has pushed");
});
