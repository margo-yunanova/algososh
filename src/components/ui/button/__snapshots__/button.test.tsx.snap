// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders button with loader 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`renders button with text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    Развернуть
  </p>
</button>
`;

exports[`renders button without text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    
  </p>
</button>
`;

exports[`renders disabled button 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  >
    Развернуть
  </p>
</button>
`;
